concent_api/core/validation.py

```diff
@@ -218,3 +225,81 @@ def validate_list_task_to_compute_ids(subtask_results_accepted_list):
+        return ('token_expiration_deadline has passed.', ErrorCode.MESSAGE_TOKEN_EXPIRATION_DEADLINE_PASSED)
+
+    # -STORAGE_CLUSTER_ADDRESS
+    if not isinstance(file_transfer_token.storage_cluster_address, str):
+        return ('Wrong type of storage_cluster_address field value.', ErrorCode.MESSAGE_STORAGE_CLUSTER_WRONG_TYPE)
+    url_validator = URLValidator()
+    try:
+        url_validator(file_transfer_token.storage_cluster_address)
+    except ValidationError:
+        return ('storage_cluster_address is not a valid URL.', ErrorCode.MESSAGE_STORAGE_CLUSTER_INVALID_URL)
+    if file_transfer_token.storage_cluster_address != settings.STORAGE_CLUSTER_ADDRESS:
+        return ('This token does not allow file transfers to/from the cluster you are trying to access.', ErrorCode.MESSAGE_STORAGE_CLUSTER_WRONG_CLUSTER)
+
+    # -CLIENT_PUBLIC_KEY
+    if not isinstance(file_transfer_token.authorized_client_public_key, bytes):
+        return ('Wrong type of authorized_client_public_key field value.', ErrorCode.MESSAGE_AUTHORIZED_CLIENT_PUBLIC_KEY_WRONG_TYPE)
+    # -FILES
+    if not all(isinstance(file, dict) for file in file_transfer_token.files):
+        return ('Wrong type of files field value.', ErrorCode.MESSAGE_FILES_WRONG_TYPE)
+    transfer_token_paths_to_files = [file["path"] for file in file_transfer_token.files]
+    if len(transfer_token_paths_to_files) != len(set(transfer_token_paths_to_files)):
+        return ('File paths in the token must be unique', ErrorCode.MESSAGE_FILES_PATHS_NOT_UNIQUE)
+
+    if not all(isinstance(file["checksum"], str) for file in file_transfer_token.files):
+        return ("'checksum' must be a string.", ErrorCode.MESSAGE_FILES_CHECKSUM_WRONG_TYPE)
+    if any((len(file["checksum"]) == 0 or file["checksum"].isspace()) for file in file_transfer_token.files):
+        return ("'checksum' cannot be blank or contain only whitespace.", ErrorCode.MESSAGE_FILES_CHECKSUM_EMPTY)
+    if not all(":" in file["checksum"] for file in file_transfer_token.files):
+        return ("'checksum' must consist of two parts separated with a semicolon.", ErrorCode.MESSAGE_FILES_CHECKSUM_WRONG_FORMAT)
+    file_checksums = [tuple(file["checksum"].split(":")) for file in file_transfer_token.files]
+    if not set(checksum_type for checksum_type, checksum in file_checksums).issubset(set(HashingAlgorithm._value2member_map_)):  # type: ignore
+        return ("One of the checksums is from an unsupported hashing algorithm.", ErrorCode.MESSAGE_FILES_CHECKSUM_INVALID_ALGORITHM)
+
+    assert set(HashingAlgorithm) == {HashingAlgorithm.SHA1}, "If you add a new hashing algorithms, you need to add validations below."
+    if any(VALID_SHA1_HASH_REGEX.fullmatch(file_checksum[1]) is None for file_checksum in file_checksums):
+        return ("Invalid SHA1 hash.", ErrorCode.MESSAGE_FILES_CHECKSUM_INVALID_SHA1_HASH)
+
+    file_sizes = [file["size"] for file in file_transfer_token.files]
```

Comment:
```
file_sizes = [file["size"] for file in file_transfer_token.files]
 +    if any((file_size is None) for file_size in file_sizes):
 +        return ("'size' must be an integer.", ErrorCode.MESSAGE_FILES_SIZE_EMPTY)
 +    for file_size in file_sizes:
 +        try:
 +            int(file_size)
 +        except (ValueError, TypeError):
 +            return ("'size' must be an integer.", ErrorCode.MESSAGE_FILES_SIZE_WRONG_TYPE)
 +
 +    if any(int(file["size"]) < 0 for file in file_transfer_token.files):
 +        return ("'size' must not be negative.", ErrorCode.MESSAGE_FILES_SIZE_NEGATIVE)
```

This can be combined into single loop.

Reply:
Done